### PR TITLE
lakeformation: Fix various bugs including SELECT permission issues

### LINKED ISSUE #20047: Lake Formation permissions: Delete problem with IAM_ALLOWED_PRINCIPALS
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.1
AWS v3.47.0

### Affected Resource

* aws_lakeformation_permissions

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_glue_catalog_database" "test" {
  name = "madagascar-database"
}

resource "aws_lakeformation_data_lake_settings" "test" {
  admins = ["arn:aws:iam::<account_id>:role/<role_name>"]
}

resource "aws_lakeformation_permissions" "test" {
  permissions = ["ALL"]
  principal   = "IAM_ALLOWED_PRINCIPALS"

  database {
    name = aws_glue_catalog_database.test.name
  }
}
```

### Debug Output

```
aws_lakeformation_data_lake_settings.test: Destroying... [id=1098952520]
aws_lakeformation_permissions.test: Destroying... [id=192198287]
aws_lakeformation_data_lake_settings.test: Destruction complete after 1s
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 10s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 20s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 30s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 40s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 50s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m0s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m10s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m20s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m30s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m40s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 1m50s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m0s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m10s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m20s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m30s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m40s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 2m50s elapsed]
aws_lakeformation_permissions.test: Still destroying... [id=192198287, 3m0s elapsed]
╷
│ Error: unable to revoke LakeFormation Permissions (input: {
│   Permissions: ["ALL"],
│   PermissionsWithGrantOption: [],
│   Principal: {
│     DataLakePrincipalIdentifier: "IAM_ALLOWED_PRINCIPALS"
│   },
│   Resource: {
│     Database: {
│       CatalogId: "<account id>",
│       Name: "madagascar-database"
│     }
│   }
│ }): InvalidInputException: No permissions revoked. Grantee does not does not have:[ALL]
```
### Expected Behavior

Resources deleted.

### Actual Behavior

1. It reports an error on `destroy`
2. Does not the delete the dependent resource `aws_glue_catalog_database`

### Steps to Reproduce

1. `terraform apply`
2. `terraform destroy`

### References

* #20048

### LINKED ISSUE #20048: Lake Formation permissions: Permission changes not forcing new with IAM_ALLOWED_PRINCIPALS
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.1
+ provider registry.terraform.io/hashicorp/aws v3.47.0

### Affected Resource(s)

* aws_lakeformation_permissions

### Terraform Configuration Files

Adding `IAM_ALLOWED_PRINCIPALS` permissions has no specific benefit but mimics what happens automatically in some accounts (e.g., not upgraded to Lake Formation model). Also, this can be done manually by practitioners. As such, it would be ideal if the AWS provider can work seamlessly with or without these extra permissions. Whether that is possible remains to be seen.

```hcl
resource "aws_glue_catalog_database" "test" {
  name = "madagascar-database"
}

resource "aws_glue_catalog_table" "test" {
  name          = "barbados-table"
  database_name = aws_glue_catalog_database.test.name

  storage_descriptor {
    columns {
      name = "rating"
      type = "DOUBLE"
    }

    columns {
      name = "year"
      type = "STRING"
    }

    columns {
      name = "movie_title"
      type = "STRING"
    }
  }
}

resource "aws_lakeformation_data_lake_settings" "test" {
  admins = ["arn:aws:iam::<account_id>:role/<role_name>"]
}

resource "aws_iam_role" "test" {
  name = "odewissug-svalbard"
  path = "/"

  assume_role_policy = jsonencode({
    Statement = [{
      Action = "sts:AssumeRole"
      Effect = "Allow"
      Principal = {
        Service = "glue.amazonaws.com"
      }
    }]
    Version = "2012-10-17"
  })
}

# this permission replicates what AWS does in accounts that haven't (or can't?) be upgraded to Lake Formation security model
# https://docs.aws.amazon.com/lake-formation/latest/dg/upgrade-glue-lake-formation.html
resource "aws_lakeformation_permissions" "iap1" {
  permissions = ["ALL"]
  principal   = "IAM_ALLOWED_PRINCIPALS"

  database {
    name = aws_glue_catalog_database.test.name
  }
}

# this permission replicates what AWS does in accounts that haven't (or can't?) be upgraded to Lake Formation security model
# https://docs.aws.amazon.com/lake-formation/latest/dg/upgrade-glue-lake-formation.html
resource "aws_lakeformation_permissions" "iap2" {
  permissions = ["ALL"]
  principal   = "IAM_ALLOWED_PRINCIPALS"

  table {
    database_name = aws_glue_catalog_database.test.name
    name          = aws_glue_catalog_table.test.name
  }
}

resource "aws_lakeformation_permissions" "test" {
  permissions = ["SELECT"]
  principal   = aws_iam_role.test.arn

  table_with_columns {
    database_name = aws_glue_catalog_database.test.name
    name          = aws_glue_catalog_table.test.name
    column_names  = ["rating", "year"]

  }
}
```

### Debug Output

```
Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

### Expected Behavior

Permissions should be updated to fewer columns.

If it were doing `ForceNew`, it would look like this:

```
Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
```

### Actual Behavior

You can add column permissions but you can't remove them:

* ✔️ `rating`, `year` 👉  `rating`, `year`, `movie_title`
* 🚫 `rating`, `year`, `movie_title` 👉  `rating`, `year`

### Steps to Reproduce

1. `terraform apply`
2. Check console for permissions and principal `odewissug-svalbard` will now have 2 columns
2. Change `aws_lakeformation_permissions.test.table_with_columns.column_names` to `["rating", "year", "movie_title"]`
3. `terraform apply`
4. Check console for permissions and principal `odewissug-svalbard` will now have the 3 columns
2. Change `aws_lakeformation_permissions.test.table_with_columns.column_names` to `["rating"]`
3. `terraform apply`
4. Check console for permissions and principal `odewissug-svalbard` will still have the 3 columns but should be 1

### Workaround

By manually removing the `IAM_ALLOWED_PRINCIPALS` permissions from affected tables and databases (e.g., using the AWS console), I have been able to get the AWS provider to correctly manage permissions.

### Important Factoids

1. Without the `IAM_ALLOWED_PRINCIPALS` permissions, the provider works as intended
2. Without the `IAM_ALLOWED_PRINCIPALS` permissions, Terraform's plan includes `~ column_names = [ # forces replacement` and `Apply complete! Resources: 1 added, 0 changed, 1 destroyed.`
2. With the `IAM_ALLOWED_PRINCIPALS` permissions, Terraform's plan does not include `~ column_names = [ # forces replacement` and shows `Apply complete! Resources: 1 added, 0 changed, 0 destroyed.`
3. In other words, with the `IAM_ALLOWED_PRINCIPALS` permissions, the AWS provider is not doing forced replacement as it should, which is why the number of permissions cannot be reduced but only increased. Decreasing requires revoking existing permissions and then granting the new permissions (which is what happens without `IAM_ALLOWED_PRINCIPALS`).

### References

* #20047

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #20047
Closes #20048

### Problems Fixed

* State not refreshing properly when using SELECT by itself in permissions involving tables with columns/tables and wildcards (AWS translates SELECT permissions from one type on input to another on output) - see details below
* Timeout and error when deleting IAP permissions
* Acceptance tests not able to run under STS credentials

### SELECT with wildcard

* The resource to which a `SELECT` permission pertains can be different than other permissions.
* Example, `SELECT` on a `table_with_columns` where `wildcard = true` is equivalent to `SELECT` on a `table`. If you input one or the other, you may get a different version back. The AWS provider did not previously handle both possibilities.
* Example, `SELECT` on a `table_with_columns` where `wildcard = true` and `name = "ALL_TABLES"` is equivalent to `SELECT` on a `table` with `wildcard = true`. The AWS provider did not previously handle both possibilities.
* This was discovered because of the incompatibility between `IAM_ALLOWED_PRINCIPALS` and individual permissions. Having `IAM_ALLOWED_PRINCIPALS` in addition to individual permissions changes the permissions. `SELECT` on a column becomes `SELECT` on wildcard, which the AWS provider did not handle correctly when refreshing state.
* However, it still remains that `IAM_ALLOWED_PRINCIPALS` is _not_ compatible with individual permissions. These changes will allow the provider to error more gracefully in that situation but not overcome the inherent incompatibility.

Output from acceptance testing (`us-west-2`):

```
--- PASS: TestAccAWSLakeFormation_serial (816.03s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable (245.57s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/iamAllowed (46.62s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/implicit (24.22s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/multipleRoles (26.05s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectOnly (24.98s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectPlus (25.03s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/basic (25.42s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectOnly (25.43s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectPlus (25.32s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardNoSelect (22.50s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns (160.48s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/basic (60.02s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/implicit (23.92s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardExcludedColumns (25.65s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectOnly (25.59s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectPlus (25.30s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (43.87s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (10.46s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (11.97s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (10.88s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (10.56s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic (230.00s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/basic (29.83s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/database (22.25s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseIAMAllowed (43.26s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseMultiple (22.56s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/dataLocation (27.14s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/disappears (84.96s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource (136.12s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/basic (23.38s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/database (27.02s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/dataLocation (29.17s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/table (29.89s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/tableWithColumns (26.67s)
```

### CODE REVIEW

File: website/docs/r/lakeformation_permissions.html.markdown [hunk 1]
```diff
@@ -10,8 +10,83 @@ description: |-
...
 
+!> **WARNING:** Lake Formation permissions are not in effect by default within AWS. Using this resource will not secure your data and will result in errors if you do not change the security settings for existing resources and the default security settings for new resources. See [Default Behavior and `IAMAllowedPrincipals`](#default-behavior-and-iamallowedprincipals) for additional details.
+
 ~> **NOTE:** In general, the `principal` should _NOT_ be a Lake Formation administrator or the entity (e.g., IAM role) that is running Terraform. Administrators have implicit permissions. These should be managed by granting or not granting administrator rights using `aws_lakeformation_data_lake_settings`, _not_ with this resource.
 
+## Default Behavior and `IAMAllowedPrincipals`
+
+**_Lake Formation permissions are not in effect by default within AWS._** `IAMAllowedPrincipals` (i.e., `IAM_ALLOWED_PRINCIPALS`) conflicts with individual Lake Formation permissions (i.e., non-`IAMAllowedPrincipals` permissions), will cause unexpected behavior, and may result in errors.
+
+When using Lake Formation, you need to choose between these two mutually exclusive options:
```
Comment (maryelizbeth) [inline website/docs/r/lakeformation_permissions.html.markdown:21]:
```suggestion
When using Lake Formation, choose ONE of the following options as they are mutually exclusive:
```

File: website/docs/r/lakeformation_permissions.html.markdown [hunk 2]
```diff
@@ -10,8 +10,83 @@ description: |-
...
+
+1. Use this resource (`aws_lakeformation_permissions`), change the default security settings using [`aws_lakeformation_data_lake_settings`](/docs/providers/aws/r/lakeformation_data_lake_settings.html), and remove existing `IAMAllowedPrincipals` permissions
+2. Use `IAMAllowedPrincipals` and not use this resource
```
Comment (maryelizbeth) [inline website/docs/r/lakeformation_permissions.html.markdown:24]:
```suggestion
2. Use `IAMAllowedPrincipals` without `aws_lakeformation_permissions` ```

File: website/docs/r/lakeformation_permissions.html.markdown [hunk 3]
```diff
@@ -10,8 +10,83 @@ description: |-
...
+This is Lake Formation's default behavior:
+
+* Lake Formation grants `Super` permission to `IAMAllowedPrincipals` on all existing AWS Glue Data Catalog resources.
+* Lake Formation enables "Use only IAM access control" for new Data Catalog resources.
+
+For more details, see [Changing the Default Security Settings for Your Data Lake](https://docs.aws.amazon.com/lake-formation/latest/dg/change-settings.html).
+
+### Problem Using `IAMAllowedPrincipals`
+
+AWS does not support combining `IAMAllowedPrincipals` permissions and non-`IAMAllowedPrincipals` permissions. Doing so results in unexpected permissions. For example, this configuration grants a user `SELECT` on a column in a table.
```
Comment (maryelizbeth) [inline website/docs/r/lakeformation_permissions.html.markdown:53]:
```suggestion
AWS does not support combining `IAMAllowedPrincipals` permissions and non-`IAMAllowedPrincipals` permissions. Doing so results in unexpected permissions and behaviors. For example, this configuration grants a user `SELECT` on a column in a table.
```

**Review by maryelizbeth — COMMENTED** [review]
Added some suggestions to the docs to enhance the flow when reading through; not essential so feel free to discard!

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSLakeFormation_serial'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLakeFormation_serial -timeout 180m
=== RUN   TestAccAWSLakeFormation_serial
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/basic
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/disappears
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/disappears
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/database
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/databaseIAMAllowed
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/databaseMultiple
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/dataLocation
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/database
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/dataLocation
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/table
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/tableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/multipleRoles
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/selectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/selectPlus
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectPlus
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/implicit
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardNoSelect
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/iamAllowed
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/implicit
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardExcludedColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectPlus
--- PASS: TestAccAWSLakeFormation_serial (937.37s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (47.04s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (12.18s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (12.18s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (11.32s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (11.36s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic (242.74s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/disappears (86.25s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/basic (33.59s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/database (24.93s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseIAMAllowed (44.04s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseMultiple (24.12s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/dataLocation (29.81s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource (140.78s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/basic (23.75s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/database (30.81s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/dataLocation (27.30s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/table (31.73s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/tableWithColumns (27.19s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable (334.87s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/multipleRoles (103.87s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectOnly (26.45s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectPlus (26.98s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectOnly (26.09s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectPlus (26.30s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/basic (28.35s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/implicit (26.36s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardNoSelect (23.25s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/iamAllowed (47.23s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns (171.95s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/basic (65.78s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/implicit (25.62s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardExcludedColumns (27.78s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectOnly (26.76s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectPlus (26.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	941.355s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSLakeFormation_serial'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLakeFormation_serial -timeout 180m
=== RUN   TestAccAWSLakeFormation_serial
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/basic
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/disappears
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/database
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/databaseIAMAllowed
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/databaseMultiple
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/dataLocation
=== RUN   TestAccAWSLakeFormation_serial/PermissionsBasic/disappears
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/dataLocation
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/table
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/tableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsDataSource/database
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/implicit
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/selectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/selectPlus
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectPlus
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/iamAllowed
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/multipleRoles
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTable/wildcardNoSelect
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/basic
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/implicit
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardExcludedColumns
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectOnly
=== RUN   TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectPlus
--- PASS: TestAccAWSLakeFormation_serial (878.63s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (64.18s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (17.19s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (15.97s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (15.70s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (15.32s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic (245.12s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/basic (25.42s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/database (25.42s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseIAMAllowed (47.88s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/databaseMultiple (30.63s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/dataLocation (30.40s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsBasic/disappears (85.37s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource (136.04s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/dataLocation (30.17s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/table (28.96s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/tableWithColumns (24.16s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/basic (24.89s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsDataSource/database (27.86s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable (255.33s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/implicit (26.19s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectOnly (25.91s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/selectPlus (25.47s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectOnly (25.25s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardSelectPlus (25.32s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/basic (27.15s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/iamAllowed (49.50s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/multipleRoles (26.26s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTable/wildcardNoSelect (24.28s)
    --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns (177.97s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/basic (70.54s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/implicit (30.45s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardExcludedColumns (25.77s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectOnly (25.61s)
        --- PASS: TestAccAWSLakeFormation_serial/PermissionsTableWithColumns/wildcardSelectPlus (25.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	881.703s
```

**Pull request merged**